docs/developer/crd-versioning.md

```diff
@@ -0,0 +1,46 @@
+# CRD Versioning
+
+This document discusses how we version our CustomResourceDefinitions (CRDs) that define Policies, Filters, or other configuration objects. NGINX Gateway Fabric distributes a few CRDs, such as NginxGateway, NginxProxy, ClientSettingsPolicy, and ObservabilityPolicy. More will inevitably be added over time.
+
+## Initial Version
+
+When creating a new CRD, the initial version should be `v1alpha1`. This version indicates a new API that needs time for use and feedback. By starting at the alpha version, it also allows us a little more flexibility in changing the API.
+
+## Changing an API
+
+When changing an API in a CRD, it's important to understand the impact of those changes. This will determine if and how we need to update the API version.
+
+**No change in version required**
+
+- Adding new optional fields.
+
+**Compatible changes, requires a version change**
+
+- When making schema changes (in other words, renaming or changing the structure of an existing field), a change of version is required (`v1alpha` to `v1alpha2`) and a [conversion webhook](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#webhook-conversion) is utilized to preserve compatibility for a user and allow our controller to support both versions.
+  - As part of this, the old API version should be marked as deprecated, and then removed after 3 releases.
+  - The new API should not be marked as the preferred or `stored` version until old API is removed.
+  - To reduce complexity, we should only support a maximum of 2 API versions at the same time.
+
+**Breaking changes, requires a version change**
+
+The following API changes are incompatible with previous API versions, and therefore not only require a version bump, but cannot use a conversion webhook. These types of changes should be avoided if at all possible due to the disruption for users. A user will need to update their configurations when upgrading NGF. These types of changes need clear messaging in release notes and docs.
+
+We'll allow a bit more flexibility for this case when dealing with alpha APIs.
+
+- Adding new required fields.
+- Removing fields. If possible, we should try to deprecate fields for 3 releases to slowly phase them out before removing.
+- Changing validation and allowed values of a field.
+
+For more in depth information on how to handle API changes, see the Kubernetes [API changes doc](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md).
+
+
+NGF version should not need to change when CRD versions change. CRDs are not embedded into NGF; they have their own versions. NGF releases are directly tied to the CRD version they support and users are required to upgrade CRD versions when upgrading NGF. This ensures that the CRD version is supported by NGF. However, the user may need to fix their CRDs if they aren't compatible.
+
+
+## Graduating to v1 (stable)
```

Comment:
Why are we jumping right from alpha to v1?

Reply:
This was following the Gateway API pattern. We certainly don't have to, but are our APIs going to be worth going through 3 stages of API versioning?

Reply:
Probably not. Can we add a note in this doc explaining why we are skipping beta?